src/api/route-services/access/roles.js

```diff
@@ -0,0 +1,46 @@
+const OWNER = 'owner';
+const ADMIN = 'admin';
+const EXPLORER = 'explorer';
+const VIEWER = 'viewer';
+
+const ROLE_NAMES = [OWNER, ADMIN, EXPLORER, VIEWER];
+
+// allowedResources contains which resources each roles is allowed to modify (read-write)
+// all the resources are viewable by default by all roles
+const allowedResources = {
+  [OWNER]: [
+    '.*',
+  ],
+  [ADMIN]: [
+    '.*',
+  ],
+  [EXPLORER]: [
+    'socket',
+    '/experiments/(?<experimentId>.*)/plots-tables/(?<plotUuid>.*)',
+    '/experiments/(?<experimentId>.*)/cellSets',
+  ],
+  [VIEWER]: [],
+};
+const isRoleAuthorized = (role, resource, method) => {
+  // if no valid role is provided, return not authorized
+  if (!ROLE_NAMES.includes(role)) {
+    return false;
+  }
+
+  // all roles have view permissions by default
+  if (method === 'GET') {
+    return true;
```

Comment:
what would this return if a user tries to view an experiment that's not theirs?

Reply:
Or would this case be ruled out by the `canAccessExperiment` function?

Reply:
that's not possible because the `role` is obtained using the `experimentId` and `userId`